self-core-impl/src/test/java/com/selfxdsd/core/managers/StoredProjectManagerTestCase.java

```diff
@@ -125,6 +125,64 @@ public void returnsProjects() {
         );
     }
 
+    /**
+     * PmUser returns its username.
+     */
+    @Test
+    public void pmUserReturnsUsername() {
+        final ProjectManager manager = Mockito.mock(ProjectManager.class);
+        Mockito.when(manager.username()).thenReturn("zoeself");
+        final User pmUser = new StoredProjectManager.PmUser(manager);
+        MatcherAssert.assertThat(
+                pmUser.username(),
+                Matchers.is("zoeself"));
+    }
+
+
+    /**
+     * PmUser returns its role.
+     */
+    @Test
+    public void pmUserReturnsRole() {
+        final User pmUser = new StoredProjectManager.PmUser(
+                Mockito.mock(ProjectManager.class)
+        );
+        MatcherAssert.assertThat(
+                pmUser.role(),
+                Matchers.is("user"));
+    }
+
+    /**
+     * PmUser returns its assigned projects.
+     */
+    @Test
+    public void pmUserReturnsProjects() {
+        final Projects assigned = Mockito.mock(Projects.class);
+        final ProjectManager manager = Mockito.mock(ProjectManager.class);
+        final User pmUser = new StoredProjectManager.PmUser(manager);
+        Mockito.when(pmUser.projects()).thenReturn(assigned);
```

Comment:
@HoreaPetrila This is not correct because PmUser.projects() is the method being tested. The correct way is:

```java
final Projects assigned = Mockito.mock(Projects.class);
final ProjectManager manager = Mockito.mock(ProjectManager.class);
Mockito.when(manager.projects).thenReturn(assigned);//instruct the manager mock to return the Projects
final User pmUser = new StoredProjectManager.PmUser(manager);
//...
```